### DISCUSSION TITLE
Could not load type 'MySql.Data.MySqlClient.MySqlConnection' from assembly 'MySqlConnector

### DISCUSSION THREAD

**Discussion by MikeAlhayek** [OrchardCMS/OrchardCore#7765] [Q&A] [answered]
I have a project written on the top on .NET 5 framework. In this project I want to to use my own UnitOfWork and Repositories along with YesSQL which is used by the CMS. I am connection to MySQL version 8.0.20 server and EntityFrameworkCore .Net 5 framework as for my ORM. To connect to MySQL, I need `Pomelo.EntityFrameworkCore.MySql -Version 5.0.0-alpha.2` which requires `MySQLConnectior >=1.1.0`

I created a project called `CoreApps.Data.Entity` and installed `OrchardCore.Data.Abstraction 1.0.0-rc2-15527`, `MySqlConnector 1.1.0` and `Pomelo.EntityFrameworkCore.MySql 5.0.0-alpha.2`

Then, I added this extension method that would configure my DbContect based on the connection string used by YesSQL

```
    public static class DependencyInjectionExtensions
    {
        public static void AddEntityDbContext(this IServiceCollection services)
        {
            services.AddDbContext<ApplicationContext>((serviceProvider, builder) =>
            {
                IDbConnectionAccessor accessor = serviceProvider.GetRequiredService<IDbConnectionAccessor>();
                IStore store = serviceProvider.GetRequiredService<IStore>();

                DbConnection connection = accessor.CreateConnection();

                if (store.Dialect.Name.Equals("MySql, StringComparison.OrdinalIgnoreCase))
                {
                    builder.UseMySql(connection, new MySqlServerVersion(connection.ServerVersion));
                } 
                else if (store.Dialect.Name.Equals("SqlServer", StringComparison.OrdinalIgnoreCase))
                {
                    
                    builder.UseSqlServer(connection;
                }
                //....
            });
        }
    }
```

But when my project starts, I get the following error

```
TypeLoadException: Could not load type 'MySql.Data.MySqlClient.MySqlConnection' from assembly 'MySqlConnector, Version=1.0.0.0, Culture=neutral, PublicKeyToken=d33d3e53aa5f8c92'.
```

MySqlConnector 1.0.0 changed its namespaces from `MySql.Data.MySqlClient` to `MySqlConnector`. I am guessing the issue is because YesSQL is using old version of MySQLConnector with the old namespaces. 

How can I workaround this issue?

**Comment by hishamco** [comment, marked as answer]
Fixed in https://github.com/sebastienros/yessql/pull/297

**Reply by MikeAlhayek** [reply]
Let's hope a new release of `OrchardCore.Data.Abstractions` becomes available soon... I am stuck :)

**Reply by hishamco** [reply]
I will try to push YesSql upgrade soon

**Reply by hishamco** [reply]
FYI still not published in NuGet I will try to take care of it once it's LIVE

**Reply by hishamco** [reply]
@CrestApps fixed in https://github.com/OrchardCMS/OrchardCore/pull/7822

**Reply by MikeAlhayek** [reply]
@hishamco  was this not yet published to the nuget server? I just updated all of the packages but still getting the same error. Was this fix included in the latest OrchardCore packages?

**Reply by hishamco** [reply]
YesSQL 1.0.0-beta-1620 version there from 4 days https://www.nuget.org/packages?q=YesSql

**Reply by jtkech** [reply]
Yes, and this last version of YesSql in only referenced by the dev branch, not by the master branch

So if you reference OrchardCore as packages, you need to use the cloudsmith feed in place of nuget

    add key="OrchardCore" value="https://nuget.cloudsmith.io/orchardcore/preview/v3/index.json" />

**Reply by hishamco** [reply]
Exactly it's for nightly build

**Reply by MikeAlhayek** [reply]
I am actually using the CloudSmith server to install my packages.  The `1.0.0-rc2-15559` version fixed the issue! Thanks a lot!

**Reply by hishamco** [reply]
All the best
